### PR TITLE
another dump of random doctests

### PR DESCRIPTION
**Pull request by KristofferC**
Passes locally excluding line numbers and https://github.com/JuliaLang/julia/pull/20288.

### CODE REVIEW

File: base/complex.jl [hunk 1]
```diff
@@ -179,6 +194,10 @@ end
     conj(z)
 
 Compute the complex conjugate of a complex number `z`.
+
+```julia
+julia> conj(1 + 3im)
+1 - 3im
```
Comment (KristofferC) [inline base/complex.jl]: miss backtic

File: base/rational.jl [hunk 2]
```diff
@@ -23,6 +23,13 @@ end
     //(num, den)
 
 Divide two integers or rational numbers, giving a `Rational` result.
+
+```jldoctest
+julia> 3 // 5
+3//5
+
+julia> (3 // 5) // (2 // 1)
+3//10
```
Comment (KristofferC) [inline base/rational.jl]: miss backtick

File: base/util.jl [hunk 3]
```diff
@@ -186,6 +218,17 @@ returning the value of the expression.
...
+```julia
+julia> @time rand(10^6);
+  0.001525 seconds (7 allocations: 7.630 MiB)
+
+julia> @time begin
+           sleep(0.3)
+           1+1
+       end
+  0.301395 seconds (8 allocations: 336 bytes)
+2
```
Comment (KristofferC) [inline base/util.jl]: ...

File: base/util.jl [hunk 4]
```diff
@@ -256,6 +313,11 @@ for the effects of compilation.
 
 See also [`@time`](@ref), [`@timev`](@ref), [`@timed`](@ref),
 and [`@elapsed`](@ref).
+
+```julia
+julia> rand(10^6)
```
Comment (KristofferC) [inline base/util.jl]: wat

File: base/complex.jl (hunk 1 above)
Comment (kshyatt) [inline base/complex.jl]: Why isn't this a doctest?

Comment (KristofferC) [inline base/complex.jl]: No reason,  will fix when back at computer.

File: base/atomics.jl [hunk 5]
```diff
@@ -88,10 +108,21 @@ function atomic_cas! end
 
 Atomically exchange the value in `x`
 
-Atomically exchanges the value in `x` with `newval`. Returns the old
+Atomically exchanges the value in `x` with `newval`. Returns the old (!)
```
Comment (tkelman) [inline base/atomics.jl]: what's this for?

File: base/complex.jl (hunk 1 above)
Comment (KristofferC) [inline base/complex.jl]: done

File: base/atomics.jl (hunk 5 above)
Comment (tkelman) [inline base/atomics.jl]: Hm you're right that is used in a bunch of these, without explanation. Couldn't we just emphasize "old" with italic or bold?

Comment (KristofferC) [inline base/atomics.jl]: Done in https://github.com/JuliaLang/julia/pull/20412/commits/dd3b5d74ebc89ce22041740ca149e43e757ba196

File: base/atomics.jl [hunk 6]
```diff
@@ -166,9 +263,20 @@ function atomic_xor! end
 
 Atomically store the maximum of `x` and `val` in `x`
 
-Performs `x[] = max(x[], val)` atomically. Returns the old (!) value.
+Performs `x[] = max(x[], val)` atomically. Returns the **old** value.
 
 For further details, see LLVM's `atomicrmw min` instruction.
```
Comment (tkelman) [inline base/atomics.jl]: is this reference backwards or is llvm's naming backwards?

File: base/atomics.jl [hunk 7]
```diff
@@ -177,9 +285,20 @@ function atomic_max! end
 
 Atomically store the minimum of `x` and `val` in `x`
 
-Performs `x[] = min(x[], val)` atomically. Returns the old (!) value.
+Performs `x[] = min(x[], val)` atomically. Returns the **old** value.
 
 For further details, see LLVM's `atomicrmw max` instruction.
```
Comment (tkelman) [inline base/atomics.jl]: and here

File: base/bitarray.jl [hunk 8]
```diff
@@ -34,6 +34,19 @@ end
 
 Construct an uninitialized `BitArray` with the given dimensions.
 Behaves identically to the [`Array`](@ref) constructor.
+
+```julia
+julia> BitArray(2, 2)
+2×2 BitArray{2}:
+ false  false
```
Comment (tkelman) [inline base/bitarray.jl]: won't this be nondeterministic?

File: base/cartesian.jl [hunk 9]
```diff
@@ -114,6 +119,15 @@ end
     @nexprs N expr
 
 Generate `N` expressions. `expr` should be an anonymous-function expression.
+```jldoctest
```
Comment (tkelman) [inline base/cartesian.jl]: blank line before?

File: base/combinatorics.jl [hunk 10]
```diff
@@ -111,6 +111,23 @@ to verify that `p` is a permutation.
...
 `permute!(v,p)` for large vectors.
+
+See also [`ipermute!`](@ref)
+
+```jldoctest
+julia> A = [1, 1, 3, 4];
+
+julia> perm = [2, 4, 3, 1];
+
+julia> permute!(A, perm);
```
Comment (tkelman) [inline base/combinatorics.jl]: these usually return A, right?

File: base/bitarray.jl (hunk 8 above)
Comment (KristofferC) [inline base/bitarray.jl]: Yeah,  so it is not a doctest

File: base/subarray.jl [hunk 11]
```diff
@@ -84,6 +84,28 @@ Like [`getindex`](@ref), but returns a view into the parent array `A` with the
...
+2×2 Array{Int64,2}:
+ 1  2
+ 3  4
+
+julia> b = view(A, :, 1)
+2-element SubArray{Int64,1,Array{Int64,2},Tuple{Base.Slice{Base.OneTo{Int64}},Int64},true}:
+ 1
+ 3
+
+julia> fill!(b, 0.0)
```
Comment (tkelman) [inline base/subarray.jl]: would using an integer fill value raise fewer eyebrows?

File: base/combinatorics.jl (hunk 10 above)
Comment (KristofferC) [inline base/combinatorics.jl]: Yeah, I thought it was too verbose to show the return AND that A has been updated.

File: base/subarray.jl (hunk 11 above)
Comment (KristofferC) [inline base/subarray.jl]: yeah probably

File: base/atomics.jl (hunk 6 above)
Comment (KristofferC) [inline base/atomics.jl]: From what I can understand of the LLVM docs, this is just backwards. Will update

**Comment by KristofferC** [comment]
Pushed an update addressing review, any more comments?

File: base/subarray.jl (hunk 11 above)
Comment (tkelman) [inline base/subarray.jl]: I don't think I originally wrote this comment on this line, but github moved it here

Comment (yuyichao) [inline base/subarray.jl]: Github (used to?) assigns comment to the wrong line if you hides white space changes.

Comment (KristofferC) [inline base/subarray.jl]: I changed the missing 0.0 -> 0

File: base/combinatorics.jl [hunk 12]
```diff
@@ -196,10 +228,18 @@ invperm(a::Tuple) = (invperm([a...])...,)
 
 #XXX This function should be moved to Combinatorics.jl but is currently used by Base.DSP.
 """
-    nextprod([k_1,k_2,...], n)
+    nextprod([k_1, k_2,...], n)
 
-Next integer not less than `n` that can be written as ``\\prod k_i^{p_i}`` for integers
+Next integer greater or equal to `n` that can be written as ``\\prod k_i^{p_i}`` for integers
```
Comment (tkelman) [inline base/combinatorics.jl]: greater than or equal to

File: base/util.jl [hunk 13]
```diff
@@ -256,6 +313,11 @@ for the effects of compilation.
 
 See also [`@time`](@ref), [`@timev`](@ref), [`@timed`](@ref),
 and [`@elapsed`](@ref).
+
+```julia
+julia> @allocated rand(10^6)
```
Comment (tkelman) [inline base/util.jl]: how predictable is this one? or do you get extra first-time compilation noise?

Comment (KristofferC) [inline base/util.jl]: Not sure. Predictable on my computer but perhaps might be OS dependent. That's why I didn't make it into a test.

**Comment by tkelman** [comment]
please fix trailing whitespace (and run the check locally if you ci skip anything)

**Comment by KristofferC** [comment]
whitespace fixed, letting ci run on this one to make sure

**Comment by tkelman** [comment]
travis failure was broken doc build (same as https://travis-ci.org/JuliaLang/julia/jobs/198521689), restarted

**Comment by KristofferC** [comment]
Good to go?

File: base/atomics.jl [hunk 14]
```diff
@@ -88,10 +108,21 @@ function atomic_cas! end
 
 Atomically exchange the value in `x`
```
Comment (tkelman) [inline base/atomics.jl]: this is a bit redundant with the next sentence

**Pull request merged**